samples/visual_indicator/nodejs/bots/teamsBot.js

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TeamsActivityHandler, TeamsInfo, TurnContext, MessageFactory, CardFactory } = require("botbuilder");
+var ACData = require("adaptivecards-templating");
+const notificationCardJson = require('../resources/SendTargetNotificationCard.json');
+
+class TeamsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+        this.baseUrl = process.env.BaseUrl;
+        this.AppId = process.env.MicrosoftAppId;
+
+        // This method is invoked whenever there is any message activity in bot's chat.
+        this.onMessage(async (context, next) => {
+            var members = new Array();
+            let meetingId = context.activity.channelData.meeting.id;
+
+            if (context.activity.value == null) {
+                TurnContext.removeRecipientMention(context.activity);
+
+                if (context.activity.text.trim() == "SendTargetedNotification") {
+                    var meetingMembers = await TeamsInfo.getPagedMembers(context);
+                    let tenantId = context.activity.channelData.tenant.id;
+
+                    for (var member in meetingMembers) {
+                        let participantDetail = await TeamsInfo.getMeetingParticipant(context, meetingId, member.aadObjectId, tenantId);
+
+                        // Select only those members that present when meeting is started.
+                        if (participantDetail.meeting.inMeeting) {
+                            members.push({ id: participantDetail.user.id, name: participantDetail.user.name })
+                        }
+                    }
+
+                    // Send and adaptive card to user to select members for sending targeted notifications.
+                    await context.sendActivity({ attachments: [this.createMembersAdaptiveCard(members)] });
+                }
+                else {
+                    await context.sendActivity("Please type `SendTargetedNotification` to send In-meeting notifications.");
+                }
+            }
+            else if (context.activity.value.Type == "StageViewNotification") {
+                var adaptiveCardChoiceSet = context.activity.value.Choice;
+                var selectedMembers = adaptiveCardChoiceSet.split(",");
+                this.stageView(context, meetingId, selectedMembers);
+            }
+            else if (context.activity.value.Type == "VisualIndicator") {
+                var adaptiveCardChoiceSet = context.activity.value.Choice;
+                var selectedMembers = adaptiveCardChoiceSet.split(",");
+                this.visualIndicator(context, meetingId, selectedMembers);
+            }
+            await next();
+        });
+    };
+
+    // Custom method for sending targeted meeting notifications in stage.
+    async stageView(context, meetingId, selectedMembers) {
```

Comment:
Do we also want to have targeted meeting notification in this sample?

Reply:
Not required, but the only purpose of adding is that user can check/test both the scenario at one place.